### PR TITLE
Move `is_min_const_fn` query to librustc_mir.

### PR DESCRIPTION
**Pull request by cjgillot**
The only two uses of the associated methods are in `librustc_mir` and 
`librustdoc`. Please tell me if there is a better choice.

cc #65031

### CODE REVIEW

File: src/librustc_mir/constness.rs [hunk 1]
```diff
@@ -1,16 +1,28 @@
...
+    fn is_const_fn(self, def_id: DefId) -> bool;
+
+    /// Whether the `def_id` is an unstable const fn and what feature gate is necessary to enable it
+    fn is_unstable_const_fn(self, def_id: DefId) -> Option<Symbol>;
+
+    /// Returns `true` if this function must conform to `min_const_fn`
+    fn is_min_const_fn(self, def_id: DefId) -> bool;
+}
+
+impl<'tcx> TyCtxtConstExt for TyCtxt<'tcx> {
```
Comment (Centril) [inline src/librustc_mir/constness.rs]: I think these can just be free functions.

Comment (oli-obk) [reply src/librustc_mir/constness.rs]: You can't add free functions to `TyCtxt` outside `librustc`

Comment (oli-obk) [reply src/librustc_mir/constness.rs]: Yea that seems better than an extension trait.

File: src/librustc_mir/constness.rs [hunk 2]
```diff
@@ -1,16 +1,28 @@
-use crate::hir;
```
Comment (oli-obk) [inline src/librustc_mir/constness.rs]: Please move this file to `src/librustc_mir/const_eval/fn_queries.rs` and reexport the functions in the `const_eval` module

**Comment by cjgillot** [comment]
Rebased with review comments.

**Comment by oli-obk** [comment]
@bors r+ rollup

Thanks!

**Pull request merged**